src/formats/SMI.cpp

```diff
@@ -333,6 +347,7 @@ void SMIFormat::read_next(Frame& frame) {
                 throw format_error("SMI Reader: rings defined with '%' must be double digits");
             }
             check_ring(parse<size_t>(smiles.substr(i + 1, 2)));
+            i += 2; // this line alone fixes most of issue 303 :)
```

Comment:
:smile: what was the problem?

Reply:
The variable `i` wasn't getting updated properly. So, on the next character, it would read the following digit character as its own ring. So, `%11` would create a ring id of `11` and then create a ring id `i` on the next character, then close that ring on the next character - causing the first error. A similar explanation caused the additional bonds with `%10` through ring id `0`.

Reply:
Thanks!